web-admin-dashboard/src/components/dashboard/delete-qns/DeleteQuestionDialog.tsx

```diff
@@ -1,71 +1,81 @@
-import { useState, useContext, useEffect } from 'react';
-import {AlertDialogAction, AlertDialogCancel, AlertDialogDescription, AlertDialogFooter, AlertDialogHeader, AlertDialogTitle, } from '@/components/ui/alert-dialog';
-import LocalQuestionRepository from '@/questionrepo/LocalQuestionRepository';
-import { useToast } from '@/components/ui/use-toast';
-import { IsChangedContext } from '@/context/IsChangedContext';
-import { Question } from '@/questionrepo/question.model';
+import { useState, Dispatch, SetStateAction } from "react";
+import {
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+} from "@/components/ui/alert-dialog";
+import { useToast } from "@/components/ui/use-toast";
+import { Question } from "@/questionrepo/question.model";
+import LiveQuestionRepository from "@/questionrepo/LiveQuestionRepository";
 
 interface Props {
   question: Question;
   setOpen: (open: boolean) => void;
+  setIsChanged: Dispatch<SetStateAction<boolean>>;
 }
 
-function DeleteQuestionDialog({ question, setOpen }: Props) {
-  const { setIsChanged } = useContext(IsChangedContext);
+function DeleteQuestionDialog({ question, setOpen, setIsChanged }: Props) {
   const [openDialog, setOpenDialog] = useState(false);
-  const [err, setError] = useState('');
+  const [err, setError] = useState("");
 
   const { toast } = useToast();
 
   const onDelete = async () => {
     try {
       // Use the LocalQuestionRepository to delete the question
-      const isDeleted = LocalQuestionRepository.deleteQuestion(question.qId);
+      // const isDeleted = LocalQuestionRepository.deleteQuestion(
+      //   question.questionId
+      // );
+      const isDeleted = await LiveQuestionRepository.deleteQuestion(
```

Comment:
You could a context to provide the repository instead of calling it statically like that. This way, you can provide any repository that implements a common interface.

In fact, the repository methods should not be static to begin with, since it needs to read .env variables at runtime.

But one point to consider is that you will be using two repositories alongside each other, one for persisting to DB and one for LocalStorage.

I recommend creating a QuestionService that handles this complex caching/fetching logic, and implements the same interface with lets say `createQuestion()` method. Then the frontend just worries about the right method to call, the Service does all the heavy lifting. Then you just have to provide QuestionService via context.